### ISSUE TITLE
DN part in training 

### ISSUE THREAD

**Issue by JosephChenHub** [IDEA-Research/GroundingDINO#135]
Hi,  I am wondering how the DN part used in training stage.  could you provide more information ?  
For DINO, "prepare_for_cdn"  is used to encode label & bbox,  and the label is encoded by a learnable "nn.Embedding''. In GroundDINO, the text input part can contain positive and negative labels,  so what is the modification of "prepare_for_cdn" to encode label ?

**Comment by SlongLiu** [comment]
The simplest solution is to use random (content) embeddings as inputs, since we have no label information now. The other parts follow the implementations in the original DINO. 

There might be other ways to deal with content embeddings for DN. But I think performance is close with large-scale training data.

**Comment by JosephChenHub** [comment]
> The simplest solution is to use random (content) embeddings as inputs, since we have no label information now. The other parts follow the implementations in the original DINO.
> 
> There might be other ways to deal with content embeddings for DN. But I think performance is close with large-scale training data.

do you mean the "prepare_for_cdn" is as same as used in DINO ?  how was the label noise obtained ?  the d_model is 256, and embedding from BERT is 768. so to adding noise, do we just randomly sample words and use the BERT embedding with a MLP (768->256) ?
